copilot-instructions.md

```diff
@@ -0,0 +1,124 @@
+
+## Pull Request Title Requirements
+
+All pull request titles must follow [Conventional Commits](https://www.conventionalcommits.org/) format to indicate the type of release. Examples:
+
+* `feat: Add new analyzer for detecting unused variables`
+* `fix: Correct false positive in PH2147 analyzer`
+* `docs: Update analyzer documentation`
+* `refactor: Simplify analyzer registration logic`
+* `test: Add coverage for edge case scenarios`
+* `chore: Miscellaneous cleanup not changing behavior`
+* `ci: Change to pipeline`
+
+**Invalid example**: `"Fix PH2147: Create new analyzer to avoid variables named exactly "_""`
+**Valid example**: `"fix: Create new analyzer PH2147 to avoid variables named exactly "_""`
+
+## Performance Considerations
+
+Roslyn analyzers run during compilation and must be performant. Follow these guidelines:
+
+* **String comparisons first**: Always perform inexpensive string comparisons of identifiers before loading the semantic model
+* **Cache expensive operations**: Store results of costly computations when possible
+* **Early returns**: Exit analysis methods as soon as you determine no issues exist
+* **Avoid regex in hot paths**: Use string methods over regular expressions for simple patterns
+
+Example:
+```csharp
+// Good: Check name first
+if (variableDeclarator.Identifier.ValueText != "_")
+    return;
+
+var semanticModel = context.SemanticModel; // Only load if needed
+
+// Bad: Load semantic model for every variable
+var semanticModel = context.SemanticModel;
+var symbol = semanticModel.GetDeclaredSymbol(variableDeclarator);
+if (symbol.Name != "_")
+    return;
+```
+
+## Documentation Requirements
+
+All analyzers must have corresponding documentation files in the [Documentation](./Documentation/) folder:
+
+* Create or update relevant documentation files (e.g., `Maintainability.md`)
+* Include rule descriptions, examples of violations, and correct usage
+* Follow the existing documentation format and style
+* Link to documentation from the main README.md if adding new analyzer categories
+
+Refer to existing documentation files for format and style guidelines.
+
+## Creating New Analyzers
+
+When creating a new analyzer, follow this workflow to ensure proper integration and testing:
+
+### Diagnostic ID Assignment
+* **Choose the next available diagnostic ID**: Review the [DiagnosticId.cs](./Philips.CodeAnalysis.Common/DiagnosticId.cs) file to find the highest existing ID and increment by 1
+* **Check open PRs**: Consider other open pull requests to avoid multiple PRs claiming the same diagnostic ID
+* **Current highest ID**: As of this writing, the highest ID is `2145`, so new analyzers should start from `2146`
+
+### Initial Configuration
+* **Set "Enabled By Default" to No**: Always set `isEnabledByDefault: false` in the `DiagnosticDescriptor` constructor
+* **Rationale**: This allows manual testing in production environments before enabling the analyzer by default
```

Comment:
Another minor item when creating new analyzers (or new files in general): the copyright at the top should use the current year. (I often see 2019 or 2022 in your work.)